frontend/src/metabase/query_builder/actions.js

```diff
@@ -988,45 +990,52 @@ export const runQuestionQuery = ({
-    // This method seems to has been a candidate for a rewrite anyway
-    const queryResult = queryResults[0];
-
-    let cardDisplay = card.display;
-
-    // try a little logic to pick a smart display for the data
-    // TODO: less hard-coded rules for picking chart type
-    const isScalarVisualization = card.display === "scalar" || card.display === "progress";
-    if (!isScalarVisualization &&
-        queryResult.data.rows &&
-        queryResult.data.rows.length === 1 &&
-        queryResult.data.cols.length === 1) {
-        // if we have a 1x1 data result then this should always be viewed as a scalar
-        cardDisplay = "scalar";
-
-    } else if (isScalarVisualization &&
-        queryResult.data.rows &&
-        (queryResult.data.rows.length > 1 || queryResult.data.cols.length > 1)) {
-        // any time we were a scalar and now have more than 1x1 data switch to table view
-        cardDisplay = "table";
-
-    } else if (!card.display) {
-        // if our query aggregation is "rows" then ALWAYS set the display to "table"
-        cardDisplay = "table";
+// TODO - the logic here is a worse version of the `guessVisualization` function in `qb/lib/actions`.
+// Why do we need two functions that do different versions of the same thing?
+//
+// try a little logic to pick a smart display for the data
+// TODO: less hard-coded rules for picking chart type
+//
+// TODO Atte Keinänen 6/1/17: Make a holistic decision based on all queryResults, not just one
+// This method seems to has been a candidate for a rewrite anyway
+const getDisplayTypeForCard = (card, queryResults, getState) => {
+    const queryResult            = queryResults[0];
+    const state                  = getState();
+    const displayAlreadyInferred = card.visualization_settings.has_inferred_display;
+    const tableMetadata          = getTableMetadata(state);
+    const innerQuery             = card.dataset_query.query;
+    const numAgs                 = (Query2.getAggregations(innerQuery) || []).length;
+    const breakoutFields         = Query2.getBreakoutFields(innerQuery, tableMetadata);
+    const numBreakouts           = (breakoutFields || []).length;
+    const isScalarViz            = card.display === "scalar" || card.display === "progress";
+    const numRows                = (queryResult.data.rows || []).length;
+    const numCols                = (queryResult.data.cols || []).length;
+
+    // if we have a 1x1 data result then this should always be viewed as a scalar
+    if (!isScalarViz && numRows === 1 && numCols === 1) return "scalar";
+
+    // timeseries. If this is the very first run and it's something that's a timeseries then switch it to line display.
+    // Otherwise if the user hits 'refresh' (displayAlreadyInferred == true) we don't want to override any changes they make to display
+    if (!displayAlreadyInferred && !isScalarViz && numAgs === 1 && numBreakouts === 1 && isDate(breakoutFields[0])) {
```

Comment:
this logic seems odd. 

maybe we should explicitly set the viz_type in the serialized query_dict in the url rather than having a magic marker? not sure ...

Reply:
I would prefer we did it that way too, but when a new question is created, its `display` type starts out as `table`, so there's no other way to tell whether a  question is a `table` because it's new or because someone manually set it to one.

Does that make sense?